commands/utils/utils_test.go

```diff
@@ -46,41 +49,168 @@ func TestChdirErr(t *testing.T) {
+	}()
+	usageGroup.Wait()
+	assert.Error(t, <-channel)
+}
+
+func getDummyRepo() RepoAggregator {
+	return RepoAggregator{
+		Repository{
+			Params: Params{
+				Git: Git{
+					RepoName: "repository1",
+				},
+			},
+		},
+		Repository{
+			Params: Params{
+				Git: Git{
+					RepoName: "repository2",
+				},
+			},
+		},
+	}
+}
+
+func TestReportEcosystemUsage(t *testing.T) {
+	const commandName = "test-command"
+	const accountName = "platform-url"
+	serverDetails := &config.ServerDetails{Url: accountName}
+	repo := getDummyRepo()
+	data, err := CreateEcosystemReports(commandName, serverDetails, repo)
+	assert.NoError(t, err)
+	assert.Len(t, data, 2)
+	for i, entry := range data {
+		assert.Equal(t, productId, entry.ProductId)
+		assert.Equal(t, accountName, entry.AccountId)
+		hashed, err := Md5Hash(repo[i].RepoName)
+		assert.NoError(t, err)
+		assert.Equal(t, hashed, entry.ClientId)
+		assert.Equal(t, commandName, entry.Features[0])
+	}
+}
+
+func TestReportEcosystemUsageError(t *testing.T) {
+	const commandName = "test-command"
+	accountName := &config.ServerDetails{Url: "platform-url/"}
+	repos := getDummyRepo()
+	// No features
+	_, err := CreateEcosystemReports("", accountName, repos)
+	assert.Error(t, err)
+	// Np repositories
+	data, err := CreateEcosystemReports(commandName, accountName, RepoAggregator{})
+	assert.NoError(t, err)
+	assert.Len(t, data, 0)
+}
+
+func TestReportXrayUsage(t *testing.T) {
+	const commandName = "test-command"
+	repo := getDummyRepo()
+	server := httptest.NewServer(createUsageHandler(t, commandName, repo[0].RepoName, repo[1].RepoName))
+	defer server.Close()
+
+	serverDetails := &config.ServerDetails{XrayUrl: server.URL + "/"}
+	var usageGroup sync.WaitGroup
+	usageGroup.Add(1)
 	channel := make(chan error)
-	go ReportUsage("", &config.ServerDetails{}, channel)
+	go func() {
+		channel <- ReportUsageToXray(commandName, serverDetails, getDummyRepo(), &usageGroup)
+	}()
+	usageGroup.Wait()
 	assert.NoError(t, <-channel)
+}
 
+func TestReportXrayError(t *testing.T) {
+	var usageGroup sync.WaitGroup
+	usageGroup.Add(1)
+	channel := make(chan error)
+	go func() {
+		channel <- ReportUsageToXray("", &config.ServerDetails{}, getDummyRepo(), &usageGroup)
+	}()
+	usageGroup.Wait()
+	assert.NoError(t, <-channel)
+
+	usageGroup.Add(1)
 	channel = make(chan error)
-	go ReportUsage("", &config.ServerDetails{ArtifactoryUrl: "http://httpbin.org/status/404"}, channel)
+	go func() {
+		channel <- ReportUsageToXray("", &config.ServerDetails{XrayUrl: "http://httpbin.org/status/404"}, getDummyRepo(), &usageGroup)
+	}()
+	usageGroup.Wait()
 	assert.Error(t, <-channel)
 }
 
-// Create HTTP handler to mock an Artifactory server suitable for report usage requests
-func createUsageHandler(t *testing.T, commandName string) http.HandlerFunc {
+// Create HTTP handler to mock an Artifactory/Xray/Ecosystem server suitable for report usage requests
```

Comment:
I couldn't find that it used for the Ecosystem server
```suggestion
// Create HTTP handler to mock an Artifactory/Xray server suitable for report usage requests
```